### ISSUE TITLE
SC2002: UUOC not caught during process substitution

### ISSUE THREAD

**Issue by devopsdeluxe** [koalaman/shellcheck#1774]
#### For bugs
- Rule Id (if any, e.g. SC1000):  [SC2002](https://github.com/koalaman/shellcheck/wiki/SC2002)

- My shellcheck version (`shellcheck --version` or "online"): 

```sh
+ shellcheck --version 
ShellCheck - shell script analysis tool
version: 0.7.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)

Unless this is something that should be [ignored](https://github.com/koalaman/shellcheck/wiki/Ignore)

- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions

- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this

- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

https://github.com/koalaman/shellcheck/issues?utf8=%E2%9C%93&q=is%3Aissue+sc2002+loop

https://github.com/koalaman/shellcheck/issues?utf8=%E2%9C%93&q=is%3Aissue+sc2002+while

https://github.com/koalaman/shellcheck/issues?utf8=%E2%9C%93&q=is%3Aissue+useless+cat+loop

https://github.com/koalaman/shellcheck/issues?utf8=%E2%9C%93&q=is%3Aissue+useless+cat+while

#### Here's a snippet or screenshot that shows the problem:

```sh
[drb@aether ~]# cat myscript
#!/usr/bin/env bash

while read -r l; do
  echo "$l"
done < foo.txt

echo

cat < foo.txt

echo

# I believe this should generate a warning:
while read -r l; do
  echo "$l"
done < <(cat foo.txt)

echo

# I believe this should generate a warning:
cat <(cat foo.txt)

echo

# I believe this should generate a warning:
cat < <(cat foo.txt)
[drb@aether ~]# cat foo.txt 
a
b
c
1
2
3
[drb@aether ~]# bash myscript 
a
b
c
1
2
3

a
b
c
1
2
3

a
b
c
1
2
3

a
b
c
1
2
3

a
b
c
1
2
3
[drb@aether ~]# shellcheck myscript
[drb@aether ~]#
```

#### Here's what shellcheck currently says:

```
```

#### Here's what I wanted or expected to see:

```sh
<(cat foo.txt)
  ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```

In my mind this is a "useless use of process substitution" rather than a "useless use of `cat`".
